src/main/java/org/apache/mesos/scheduler/repair/CoreFrameworkState.java

```diff
@@ -0,0 +1,44 @@
+package org.apache.mesos.scheduler.repair;
+
+import org.apache.mesos.Protos.FrameworkID;
+import org.apache.mesos.Protos.TaskID;
+import org.apache.mesos.Protos.TaskInfo;
+import org.apache.mesos.state.StateStoreException;
+
+import java.util.List;
+
+/**
+ * Exposes simple task-oriented API for components that need to interact with the framework
+ * as a whole.
+ */
+public interface CoreFrameworkState {
```

Comment:
Why does this interface exist?  It looks almost identical to a subset of StateStore.  It adds only `deleteTask` which I don't think belongs here, or has unclear meaning at least.  It also adds `getTerminatedTasks`.  Why not just add that to the StateStore?

`getFrameworkId` and `getTaskInfos` are redundant with `StateStore` capabilities